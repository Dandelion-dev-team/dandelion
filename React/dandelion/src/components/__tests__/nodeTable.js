import React from "react"
import renderer from "react-test-renderer"
import NodeTable from "../nodeTable"


describe("Node Table", () => {
    let originalFetch;

    beforeEach(() => {
        originalFetch = global.fetch;
        global.fetch = jest.fn(() => Promise.resolve({
            json: () => Promise.resolve({
                "nodes": {
                    "id": 1,
                    "last_communication_date": "2022-02-18T13:07:39.231Z",
                    "health_status": "green",
                    "status": "active"
                  }
            })
        }));
    });

    afterEach(() => {
        global.fetch = originalFetch;
    });

    it("renders correctly", () => {
        const tree = renderer
            .create(<NodeTable/>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})

