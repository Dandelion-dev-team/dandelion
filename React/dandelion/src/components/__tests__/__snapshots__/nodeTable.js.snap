// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth renders correctly 1`] = `
<div
  className="authTable"
>
  <table
    className="tableList"
  />
</div>
`;

exports[`Node Table renders correctly 1`] = `
<div
  className="authTable"
>
  <table
    className="tableList"
  />
</div>
`;
